src/frogger.cpp

```diff
@@ -87,12 +122,25 @@ Rcpp::List run_base_model(const Rcpp::List data,
                                          hiv_adult_first_age_group,
                                          treatment_stages,
                                          time_art_start,
+                                         adult_incidence_first_age_group,
+                                         pAG_INCIDPOP,
+                                         hiv_steps,
+                                         (1.0 / hiv_steps),
```

Comment:
save this as dt first, don't do the calculation here

Reply:
or pass hiv_steps to the constructor and compute dt in the struct. that way 1/dt is always integral